src/api/uploads/MultiputUpload.js

```diff
@@ -0,0 +1,367 @@
+        this.clientId = null;
+    }
+
+    /**
+     * Creates upload session. If a file ID is supplied, use the Chunked Upload File Version
+     * API to replace the file.
+     *
+     * @private
+     * @param {Object} - Request options
+     * @param {boolean} [options.url] - Upload URL to use
+     * @param {string} [options.fileId] - ID of file to replace
+     * @param {string} [options.fileName] - New name for file - required for new file upload
+     * @return {void}
+     */
+    createUploadSession = async (): Promise<> => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        // Set up post body
+        const postData: StringAnyMap = {
+            file_size: this.file.size,
+            file_name: this.file.name
+        };
+
+        if (this.destinationFolder) {
+            postData.folder_id = this.destinationFolder;
+        }
+
+        try {
+            const data = await this.xhr.post({ url: this.sessionEndpoints.createSessionUrl, data: postData });
+            this.uploadSessionSuccessHandler(data);
+        } catch (error) {
+            this.uploadSessionErrorHandler(error);
+        }
+    };
+
+    /**
+     * Upload session session error handler.
+     * Retries the create session request or fails the upload.
+     * TODO: implement this
+     * 
+     * @private
+     * @return {void}
+     */
+    // eslint-disable-next-line
+    uploadSessionErrorHandler = (error: Error): void => {};
+
+    /**
+     * Handles a upload session success response
+     *
+     * @private
+     * @param {Object} data - Upload session creation success data
+     * @return {void}
+     */
+    uploadSessionSuccessHandler = (data: any): void => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const { id, part_size, session_endpoints } = data;
+
+        this.sessionId = id;
+        this.partSize = part_size;
+        this.sessionEndpoints = {
+            ...this.sessionEndpoints,
+            uploadPart: session_endpoints.upload_part,
+            listParts: session_endpoints.list_parts,
+            commit: session_endpoints.commit,
+            abort: session_endpoints.abort,
+            logEvent: session_endpoints.log_event
+        };
+
+        this.populateParts();
+        this.processNextParts();
+    };
+
+    /**
+     * Attempts to process more parts, except in the case where everything is done or we detect
+     * a file change (in which case we want to abort and not process more parts).
+     * 
+     * @private
+     * @return {void}
+     */
+    processNextParts = (): void => {
+        if (this.failSessionIfFileChangeDetected()) {
+            return;
+        }
+
+        if (this.partsUploaded === this.parts.length && this.fileSha1) {
+            this.commitSession();
+        }
+
+        this.updateFirstUnuploadedPartIndex();
+
+        while (this.canStartMorePartUploads()) {
+            this.uploadNextPart();
+        }
```

Comment:
We need to kick off the digest computation for parts here too (unless you're planning to do it elsewhere or add it later)

Reply:
I'll add a TODO here